main/utils.js

```diff
@@ -13,6 +15,25 @@ function formatTokenValue (input) {
   return Number(number.toFixed(6))
 }
 
+const allowedURLs = [
+  'https://filspark.com/',
+  'https://github.com/filecoin-station/voyager',
+  'https://filstation.app/',
+  'https://github.com/filecoin-station/spark',
+  'https://beryx.zondax.ch',
+  'https://beryx.io',
+  'https://pl-strflt.notion.site'
+].map(str => new URL(str))
+
+/**
+   * @param {string} url
+*/
+function validateExternalURL (url) {
+  const normalizedURL = new URL(url).href
+  assert(allowedURLs.find(item => normalizedURL.startsWith(item.href)))
```

Comment:
I just realized `startsWith` isn't safe enough, because we can't trust all content on github for example. Could you please change it to either do full matches, or regexps?

Reply:
There's only a problem: Go to https://github.com/filecoin-station/voyager/../../facebook/react/issues/29598

Reply:
Ahh right, good catch!

Reply:
One potential fix:

```js
const url = new URL(input)
validate(url.href)
```

Using `URL`, paths are normalized automatically.

Reply:
```js
const url = new URL('https://github.com/filecoin-station/voyager/../../facebook/react/issues/29598')
url.href // => 'https://github.com/facebook/react/issues/29598'
```

Reply:
I added pattern validation, so we can have both full match or regexp validation. wdyt?